tests/test_random_models.py

```diff
@@ -0,0 +1,77 @@
+#! /usr/bin/env python
+import os
+from copy import deepcopy
+
+# import matplotlib
+# matplotlib.use('TKAgg')
+import matplotlib.pyplot as plt
+import pytest
+import numpy as np
+import astropy.units as u
+
+import pint.models as tm
+from pint import fitter, toa
+from pinttestdata import datadir
+import pint.models.parameter as param
+from pint import ls
+from pint import utils
+
+
+@pytest.mark.skipif(
+    "DISPLAY" not in os.environ, reason="Needs an X server, xvfb counts"
```

Comment:
Agree.  I had taken that from `test_fitter.py/test_fitter()` as noted, but it is not necessary.

Also agree about testing the `CovarianceMatrix` class more generally, but I mostly just made small tweaks and it would take a while to come up with all possible tests.  I can start thinking about that...